### PR TITLE
Recipe app

### PR DESCRIPTION
**Pull request by Dachrono**
# 🎉 🎈 In this *Milestone* 🙋‍♂️ 🚩 ‼ :

**Hi there** 👋

- Follow the best GitHub flow. ✅
- Follow our list of best practices for Ruby. ✅
- Document in a professional way. ✅
- No linters errors. ✅
- Create a new app. ✅
- Create a login page (use of devise). ✅
- Create a registration page. ✅
- Create a food list (with all CRUD implementation, except for 'update'). ✅
- Create a list (with all CRUD implementation, except for 'update'). ✅
- Create a general shopping list view (all missing food for all your recipes and total price). ✅
- Create a list of all public recipes from other users with their names and total prices. ✅
- Testing using Capybara. ✅

# To the code reviewer 👨‍💻
I really thank 🙇‍♂️ you, dear code reviewer 👨‍🎨 for dedicating your time ⌚ to check the PR 🧾 .

I kindly ask code reviewer 🤵 that if there is any issue in this project, please do list 📃 them in a descriptive 💡 manner and give your best suggestions if needed.

If you think some significant issues need to be changed ♻️ please kindly contact 📞 me through Zoom or Slack, I am available on Microverse UTC-6 time zone.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+If you like this project please feel free to send me corrections for make it better I would feel glad to read your comments.
+And think If you enjoy gift me a star.  
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
```
Comment (Owusu-Desmond) [inline README.md:166]: - [ ] I suggest that you acknowledge anyone that  inspire instead of leaving your acknowledge section blank :blush:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @Dachrono and @ClaudiaRojasSoto , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Great working on the set up of the project ✅
📌 Good commit messages ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (ClaudiaRojasSoto) [reply README.md:166]: Done!

File: app/views/recipes/public_recipes.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <div class="inner-padding">
+    <h1 class="header-public-recipe-list">Public Recipes</h1>
+    <ul class="recipe-list">
+      <% @public_recipes_data.each do |recipe| %>
+        <li class="recipe-item">
+          <div class="recipe-info">
+            <div class="recipe-title"><%= recipe.name %></div>
+            <div class="recipe-author">by <%= recipe.user.name %></div>
+          </div>
+          <div class="recipe-details">
```
[the remark this answers is not in the document]
Comment (ClaudiaRojasSoto) [reply app/views/recipes/public_recipes.html.erb]: Done!

File: app/controllers/recipes_controller.rb [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      foods_needed.sort_by { |food_name, _| Food.obtener_precio(food_name) }.to_h
+    when 'price_desc'
+      foods_needed.sort_by { |food_name, _| Food.obtener_precio(food_name) }.reverse.to_h
+    else
+      foods_needed
+    end
+  end
+
+  private
+
```
[the remark this answers is not in the document]
Comment (ClaudiaRojasSoto) [reply app/controllers/recipes_controller.rb:84]: @Owusu-Desmond. We resolved the other suggestions but this one is working as expected. 
List of foods (you can see the foods, quantities, and prices
Captura de pantalla 2023-10-13 a la(s) 22 45 46
You can see 3 recipes in this case...
Captura de pantalla 2023-10-13 a la(s) 22 45 54
The details of foods in every recipe of the user...
Captura de pantalla 2023-10-13 a la(s) 22 46 05
Captura de pantalla 2023-10-13 a la(s) 22 46 19
Captura de pantalla 2023-10-13 a la(s) 23 08 56
And the shopping list of the missing foods
Captura de pantalla 2023-10-13 a la(s) 23 08 46

**Review by Reem-lab — APPROVED** [review]
# Hi @ClaudiaRojasSoto and @Dachrono,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 💯
- [ ]  Descriptive PR⭐
- [ ] Great for adding the reviews to each other 🔥🔥
- [ ] Amazing working on the views, it is like the wireframe 🚀
- [ ] Good job working on the models and associations ✅
- [ ] Kudos for adding the schema with all the indexes on the foreign keys ⭐
- [ ]  Professional README 🤩
- [ ]  No Linters error ✅
- [ ]  Huge efforts working on the tests 🙏🏻💯
- [ ] Great routes for adding the needed ones ⭐

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ClaudiaRojasSoto** [comment]
Thank you @Reem-lab!

**Pull request merged**